templates/tfengine/components/cicd/variables.tf

```diff
@@ -25,3 +25,15 @@ variable "state_bucket" {
   description = "Name of the Terraform state bucket"
   type        = string
 }
+
+variable "logs_bucket" {
+  description = "Name of the bucket used for cloudbuild logs"
+  default     = "cloudbuild-sa@{{.project_id}}.iam.gserviceaccount.com"
+  type        = string
+}
+
+variable "service_account_email" {
+  description = "Service account used for cloudbuild triggers"
+  default     = "{{.project_id}}-cloudbuild-logs-bucket"
```

Comment:
The `default` values seem swapped.

Reply:
The default value for the bucket can be invalid if the project ID contains characters that bucket name doesn't like.

https://cloud.google.com/resource-manager/docs/creating-managing-projects
https://cloud.google.com/storage/docs/naming-buckets

Asking the users to provide the bucket name is probably safer. And. we only use the custom logs bucket in the google_cloudbuild_trigger resource when bucket is explicitly asked for and created, otherwise, we can leave that not specified and let cloud build trigger to use the default bucket.

Reply:
So we shouldn't create the bucket, right? Either user provides creates one and provides it here, or we use the default bucket?

Reply:
Resolving based on our conversation.